### PR TITLE
[BUG] Pokemon with the Levitate ability - Sand Attack Interaction Fix

### PR DESCRIPTION
**Pull request by frutescens**
## What are the changes?
Pokemon with Levitate are not immune to Sand Attack despite Sand Attack's Ground type because Levitate is a neutral ability and Sand Attack is a status move. In PokeRogue, these Pokemon were immune to Sand Attack because of a missing move category check. 

Link to Bug [Here](https://github.com/pagefaultgames/pokerogue/issues/2447)

## Why am I doing these changes?
Trying to help with some bugs here and there. 

## What did change?
data/ability.ts : Levitate's implementation --> TypeImmunityAbAttr now checks for a move category before determining if a move should effect the Pokemon. 

### Screenshots/Videos
https://github.com/user-attachments/assets/d79b7ee3-e0f6-4a65-a5c6-2cf9c9354c83

## How to test the changes?
Use Sand Attack on a Pokemon with Levitate

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

**Comment by Tempo-anon** [comment]
Sap sipper test broke

File: src/data/ability.ts [hunk 1]
```diff
@@ -391,6 +389,19 @@ export class TypeImmunityAbAttr extends PreDefendAbAttr {
   }
 }
 
+export class TypeImmunityNeutralAbAttr extends TypeImmunityAbAttr {
```
Comment (innerthunder) [inline src/data/ability.ts:392]: nit: Rename to `AttackTypeImmunityAbAttr`

Comment (frutescens) [reply src/data/ability.ts:392]: Good point. I'll rename it

**Review by innerthunder — APPROVED** [review]
lgtm

File: src/data/ability.ts [hunk 2]
```diff
@@ -391,6 +389,19 @@ export class TypeImmunityAbAttr extends PreDefendAbAttr {
   }
 }
 
+export class AttackTypeImmunityAbAttr extends TypeImmunityAbAttr {
```
Comment (Tempo-anon) [inline src/data/ability.ts:392]: where documentation

**Review by Tempo-anon — COMMENTED** [review]
What about sand attack on a flying type? Or Pokemon with Magnet Rise tag?

**Comment by frutescens** [comment]
Hi, Pokemon with the flying-type or with the magnet rise tag are already immune to Sand Attack.

File: src/data/ability.ts (hunk 2 above)
Comment (frutescens) [reply src/data/ability.ts:392]: ):

**Pull request merged**